### PR TITLE
Fix bug with vkCmdEndQuery and move tests

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Using Profiles/MockICD I found we have not been testing with a `VK_KHR_performance_query` device as there was a bug where we were off by one

To make future query-related testing easier, moved query tests to own file

### CODE REVIEW

File: layers/state_tracker/cmd_buffer_state.h [hunk 1]
```diff
@@ -167,7 +167,7 @@ class CMD_BUFFER_STATE : public REFCOUNTED_NODE {
     bool has_build_as_cmd;
 
     CB_STATE state;               // Track cmd buffer update state
-    uint64_t commandCount;        // Number of commands recorded. Currently only used with VK_KHR_performance_query
+    uint64_t command_count;       // Number of commands recorded. Currently only used with VK_KHR_performance_query
```
Comment (ncesario-lunarg) [inline layers/state_tracker/cmd_buffer_state.h:170]: Might be good to do variable renaming in a separate commit...or just wait until we do that impending/"some day" clang-tidy refactor.

File: tests/vklayertests_query.cpp [hunk 2]
```diff
@@ -0,0 +1,2385 @@
...
+
+#include "cast_utils.h"
+#include "core_validation_error_enums.h"
+#include "enum_flag_bits.h"
+#include "layer_validation_tests.h"
+#include "vk_layer_utils.h"
+#include "generated/vk_validation_error_messages.h"
+#include "vk_layer_logging.h"
+
+TEST_F(VkLayerTest, QueryPerformanceCreation) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_query.cpp:23]: Would it make sense to namespace these accordingly (e.g., `VkLayerQueryTest, ...`)? Maybe not for this change, but asking more in general if having a namespace per-file would make sense.

**Review by ncesario-lunarg — APPROVED** [review]

File: tests/vklayertests_query.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/vklayertests_query.cpp:23]: Had Exact thought, I don't see why not to start now

File: layers/state_tracker/cmd_buffer_state.h (hunk 1 above)
Comment (spencer-lunarg) [reply layers/state_tracker/cmd_buffer_state.h:170]: agree, can make a separate commit here
